test/common/upstream/health_check_fuzz.h

```diff
@@ -87,6 +87,31 @@ class GrpcHealthCheckFuzz : public HealthCheckFuzz, GrpcHealthCheckerImplTestBas
   // Determines whether a client closes after responds and timeouts. Exactly maps to
   // received_no_error_goaway_ in source code.
   bool received_no_error_goaway_ = false;
+
+  // In order to not affect unit tests, move state here. Since only one test session, we don't need
+  // a vector. Also, we should make Timers NiceMocks.
+  struct TestSession {
+    TestSession() = default;
```

Comment:
Is this needed?

Reply:
Deleted. In terms of "boilerplate" copy paste, do you mean shared logic across the three base classes in regards to mocks?

Reply:
I took a look around expectations on mocks. I think that TCP should be distinct, as it has a lot less state and logic as the HTTP and gRPC. However, HTTP and gRPC are almost identical. I think that you could perhaps combine HTTP and gRPC  and get rid of similar code. In regards to the timeframe for my internship and triaging work, I think I'll pass on messing with this for now and focus on metrics like you and Asra mentioned :).

Reply:
Fair enough, just wanted to check to see if there was opportunity to make this more maintainable.